### PR TITLE
OMTF: From cmssw 10 0 0 kb1

### PR DESCRIPTION
**Pull request by kbunkow**
Added second implementation of the final ghostBuster-sorter: GhostBusterPreferRefDt. It gives
better rate reduction without drop of efficiency. By default the old GhostBuster is used, and to switch to the GhostBusterPreferRefDt one have to add ghostBusterType = cms.string("GhostBusterPreferRefDt") into L1TMuonOverlapTrackProducer configuration (e.g. in simOmtfDigis_cfi.py)

In the GhostBusterPreferRefDt, if the count of the fired layers is the same, the output candidates are sorted according to the number of the reference layer. In this way, in the GhostBusterPreferRefDt the candidate based on the DT reference layer has priority over the one with the CSC or RPC reference layer. In the original version (i.e. GhostBuster) the second sorting condition is the sum of the pdfs values.

### CODE REVIEW

File: L1Trigger/L1TMuonOverlap/src/OMTFReconstruction.cc [hunk 1]
```diff
@@ -71,7 +72,15 @@ void OMTFReconstruction::beginRun(edm::Run const& run, edm::EventSetup const& iS
 
   m_OMTFConfig->configure(omtfParams);
   m_OMTF->configure(m_OMTFConfig, omtfParams);
-  m_GhostBuster.setNphiBins(m_OMTFConfig->nPhiBins());
+  //m_GhostBuster.setNphiBins(m_OMTFConfig->nPhiBins());
+
+  if(m_Config.exists("ghostBusterType") ) {
+    if(m_Config.getParameter<std::string>("ghostBusterType") == "GhostBusterPreferRefDt")
+    	m_GhostBuster.reset(new GhostBusterPreferRefDt(m_OMTFConfig) );
```
Comment (rekovic) [inline L1Trigger/L1TMuonOverlap/src/OMTFReconstruction.cc:79]: @kbunkow How is this memory dealocated

File: L1Trigger/L1TMuonOverlap/src/OMTFReconstruction.cc [hunk 2]
```diff
@@ -71,7 +72,15 @@ void OMTFReconstruction::beginRun(edm::Run const& run, edm::EventSetup const& iS
...
+  }
+  else
+	  m_GhostBuster.reset(new GhostBuster() );
```
Comment (rekovic) [inline L1Trigger/L1TMuonOverlap/src/OMTFReconstruction.cc:82]: Same here, how is this memory dealocated ?

File: L1Trigger/L1TMuonOverlap/src/OMTFReconstruction.cc (hunk 1 above)
Comment (kbunkow) [reply L1Trigger/L1TMuonOverlap/src/OMTFReconstruction.cc:79]: @rekovic 
 m_GhostBuster is std::unique_ptr<IGhostBuster> (L1Trigger/L1TMuonOverlap/interface/OMTFReconstruction.h line 77), so the memory is deallocated automatically  when OMTFReconstruction is destructed.

Comment (rekovic) [reply L1Trigger/L1TMuonOverlap/src/OMTFReconstruction.cc:79]: @kbunkow Ah, yes.

Comment (rekovic) [reply L1Trigger/L1TMuonOverlap/src/OMTFReconstruction.cc:79]: Thx!

**Comment by rekovic** [comment]
This is cherry-picked and merged in l1t-integration-CMSSW_10_1_2, tagged as `l1t-integration-v97.28`.

@kbunkow Please advise if we should already turn ON this new GhostBuster  in the config simOmtfDigis_cfi.py (before L1T PR to CMSSW). Do you plan to present new rates/eff at L1T Meeting?

**Comment by kbunkow** [comment]
@rekovic The firmware with the new GhostBuster is not yet deployed in the P5, so at least from the point of view of the DQM data-emulator comparison this should not be enabled in the emulator.
Karol
cc: @konec

**Comment by abrinke1** [comment]
Hi @kbunkow , @konec 
Shouldn't this be a parameter that is implemented in O2O?  Or in the emulator, could you make it depend on the FW version number, so it will only run with the new GhostBuster once the new FW is deployed?
Cheers, Andrew

**Comment by rekovic** [comment]
@abrinke1 Thanks, I second it. This would be a correct approach.

**Comment by rekovic** [comment]
attn @kbunkow @konec @akalinow @thomreis 

FYI, PRs #700 and #701 are merged and tagged v98.13 and v98.14 respectively.
These two PRs use the fwVersion provided in the xml file to invoke the new version of emulator to correspond to the fw used at p5.  
To be done: have the fwVersion O2Oed, so that we can get rid of the fakeOmtfParams.

Will close this PR.  Expect a separate PR with OMTF O2O.

**Pull request closed without merging**